torch/testing/_internal/common_methods_invocations.py

```diff
@@ -10511,13 +10511,15 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
                DecorateInfo(unittest.skip("Skipped!"), 'TestJit', 'test_variant_consistency_jit'),
                # RuntimeError: unsupported memory format option Preserve
                DecorateInfo(unittest.skip("Skipped!"), 'TestJit', 'test_variant_consistency_jit'),
-               # GradcheckError: gradcheck expects all tensor inputs are dense when check_sparse_nnz is set to False
+               # RuntimeError: sparse_mask does not support automatic differentiation for outputs with complex dtype
+               # RuntimeError: Sparse CSR tensors do not have strides
                DecorateInfo(unittest.skip("Skipped!"), 'TestFwdGradients', 'test_fn_fwgrad_bwgrad'),
-               # GradcheckError: gradcheck expects all tensor inputs are dense when check_sparse_nnz is set to False
+               # ValueError: Sparse output is not supported at gradcheck yet. Please call to_dense(masked_grad=...) ...
```

Comment:
Ho I didn't realize that the new errors are not `GradcheckError`. It might be best to move these from ValueError to GradcheckError for consistency (and work better with the no-error mode).

Reply:
Hmm we may actually prefer to have ValueError for when its something the caller can easily correct, e.g. by modifying the input/output types of their function or passing in different arguments.

Reply:
ok, IIUC, we'll keep this exception as ValueError.